src/SWP/Component/MultiTenancy/Factory/TenantFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher MultiTenancy Component.
+ *
+ * Copyright 2016 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Component\MultiTenancy\Factory;
+
+/**
+ * Class TenantFactory.
+ */
+class TenantFactory implements TenantFactoryInterface
+{
+    /**
+     * @var string
+     */
+    protected $className;
+
+    /**
+     * TenantFactory constructor.
+     *
+     * @param string $className
+     */
+    public function __construct($className)
+    {
+        $this->className = $className;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create()
```

Comment:
but then it means, in the code you will need to use `$tenant = TenantFactory::create(Tenant::class);` everywhere where you will want to instantiate a new tenant object. If for some reason you will change that class, its method etc. you will need to change it everywhere you used it. Why not use Symfony container? also static = evil.